api/rest/client/lbclient.go

```diff
@@ -0,0 +1,431 @@
+package client
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	cid "github.com/ipfs/go-cid"
+	shell "github.com/ipfs/go-ipfs-api"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/ipfs-cluster/api"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+// loadBalancingClient a client to interact with IPFS Cluster APIs.
+// It balances the load by distributing it among peers.
+type loadBalancingClient struct {
+	strategy LBStrategy
+}
+
+// LBStrategy is a strategy to load balance request among clients
+type LBStrategy interface {
+	Next(count int, call func(Client) error) error
+}
+
+// RoundRobin is a load balancing strategy that would use clients in a sequence.
+type RoundRobin struct {
+	clients []Client
+	length  int
+	retries int
+	counter chan int
+}
+
+// NewRoundRobin would return an LBStrategy that load balances requests by using
+// clients in sequence.
+func NewRoundRobin(cfgs []*Config, retries int) (LBStrategy, error) {
+	var clients []Client
+	for _, cfg := range cfgs {
+		defaultClient, err := NewDefaultClient(cfg)
+		if err != nil {
+			return nil, err
+		}
+		clients = append(clients, defaultClient)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+
+	counter := make(chan int)
+	counter <- 0
```

Comment:
I think this is hanging forever and this function will never return. `make(chan int, 1)` might help.

The problem with using channels is that you will leak them as you have no way of closing them when the client is not used anymore and they will be happily stay open and holding data. Consider using the stuff `atomic` for this, which is also possibly way faster.

Reply:
https://github.com/ipfs/ipfs-cluster/pull/737/commits/c24f0004fe9ccdf7d365d8709d498c2fdad5a635

Uses atomic now